lua/precognition/init.lua

```diff
@@ -323,9 +323,17 @@ function M.show()
     end
     visible = true
 
+    local prev_line = vim.api.nvim_win_get_cursor(0)[1]
     local draw = display_marks
     if config.debounce > 0 then
-        draw = utils.debounce_trailing(display_marks, config.debounce)
+        local debounced = utils.debounce_trailing(display_marks, config.debounce)
+        draw = function(...)
+            if vim.api.nvim_win_get_cursor(0)[1] == prev_line then
```

Comment:
how would this behave if you had the same identical line twice?
how would this behave if you had the same identical line twice?

Reply:
Lol my bad, requested review too early